core/cmd/shell_local.go

```diff
@@ -900,6 +923,48 @@ func (s *Shell) CreateMigration(c *cli.Context) error {
 	return nil
 }
 
+// CleanupChainTables deletes database table rows based on chain type and chain id input.
+func (s *Shell) CleanupChainTables(c *cli.Context) error {
+	cfg := s.Config.Database()
+	parsed := cfg.URL()
+	if parsed.String() == "" {
+		return s.errorOut(errDBURLMissing)
+	}
+
+	dbname := parsed.Path[1:]
+	if !c.Bool("dangerWillRobinson") && !strings.HasSuffix(dbname, "_test") {
+		return s.errorOut(fmt.Errorf("Cannot reset database named `%s`. This command can only be run against databases with a name that ends in `_test`, to prevent accidental data loss. If you really want to delete chain specific data from this database, pass in the -dangerWillRobinson option", dbname))
+	}
+
+	db, err := newConnection(cfg)
+	if err != nil {
+		return s.errorOut(errors.Wrap(err, "Error connecting to the database"))
+	}
+
+	defer db.Close()
+
+	tablesToDeleteFromQuery := "SELECT table_name FROM information_schema.columns WHERE \"column_name\"=$1;"
+	// Delete rows from each table based on the chain_id.
+	if strings.EqualFold("EVM", c.String("chainType")) {
+		var tables []string
+		if err = db.Select(&tables, tablesToDeleteFromQuery, "evm_chain_id"); err != nil {
+			return err
+		}
+		for _, tableName := range tables {
+			query := fmt.Sprintf("DELETE FROM %s WHERE \"evm_chain_id\"=$1;", tableName)
+			_, err = db.Exec(query, c.Int("id"))
```

Comment:
What about this one too? The escaped quotes are also surprising.

Reply:
Escaped quotes are used in psql to delimit identifiers, it works without but this should be fine. 

And about the table name injection, I'm not sure how else to do it, it can't be done in the same way as values are

Reply:
Ah ok. What about this format?
```go
fmt.Sprintf(`DELETE FROM %s WHERE "evm_chain_id"=$1;`, tableName)
```

Reply:
What is the difference?

Reply:
Cool, forgot that this is possible